package.json

```diff
@@ -64,21 +64,21 @@
     "@rollup/plugin-replace": "^2.0.0",
     "@types/react": "^16.0.0",
     "@types/react-dom": "^16.0.0",
+    "@wojtekmaj/enzyme-adapter-react-17": "^0.1.1",
     "babel-plugin-transform-react-remove-prop-types": "^0.4.24",
     "dtslint": "^4.0.0",
     "enzyme": "^3.10.0",
-    "enzyme-adapter-react-16": "^1.0.0",
     "eslint": "^7.0.0",
     "eslint-config-prettier": "^6.0.0",
     "eslint-config-sanity": "^4.0.0",
     "eslint-plugin-react": "^7.0.0",
     "jest": "^26.0.0",
     "npm-run-all": "^4.0.0",
     "prettier": "^2.0.0",
-    "react": "^16.0.0",
-    "react-dom": "^16.0.0",
     "react-katex": "^2.0.0",
-    "react-test-renderer": "^16.0.0",
+    "react": "^17.0.1",
+    "react-dom": "^17.0.1",
+    "react-test-renderer": "^17.0.1",
```

Comment:
```suggestion
    "react": "^17.0.0",
    "react-dom": "^17.0.0",
    "react-test-renderer": "^17.0.0",
```

Reply:
Why? Aesthetics?

Reply:
Broadening the version range as much as possible, gives package managers more opportunity to de-duplicate dependencies.
Which in turn can help bundlers like CRA which will error it detects multiple react versions (without defining their own renderer).

Reply:
I see the point for dependencies, not sure I see the big deal with dev dependencies. Also, following that logic, shouldn't the version range be >=16, then? That's broader, and strictly speaking compatible?

Reply:
I personally avoid `>=` as much as possible preferring a combination of minor ranges `^16 || ^17`.
But I know @wooorm has a different opinion on that.

Reply:
🤷  Not a hill I'm willing to die on 😉 
^17.0.0 works for me.

Reply:
Removing meaningless stuff in semver ranges helps clean up package locks, and indeed should help with deduping. And it’s just cleaner: there is really no need to be so specific, having the zeros there makes the major more pronounced, and it’s more clear for for cases that do need it (`"html-to-react": "^1.4.5"`).

For the difference between how the peer dep doesn’t have an upper boundary but dev-deps do: My experience it’s similar to `engines`: I can guarantee that it works on v16, and it will probably work afterwards as well. For example: we’re currently blocking folks from updating because there is an error, even though it works fine and react 17 didn‘t break us. and react 18 also probably won’t.